apps/total-typescript/src/ability/ability.ts

```diff
@@ -0,0 +1,59 @@
+import {Ability, AbilityBuilder, AbilityClass} from '@casl/ability'
+import {Exercise} from '../lib/exercises'
+import {SanityDocument} from '@sanity/client'
+
+type Actions = 'manage' | 'invite' | 'view'
+type Subjects =
+  | 'Team'
+  | 'Purchase'
+  | 'Content'
+  | 'Product'
+  | 'Invoice'
+  | 'Account'
+  | 'all'
+export type AppAbility = Ability<[Actions, Subjects]>
+const AppAbility = Ability as AbilityClass<AppAbility>
+
+type ViewerAbilityInput = {
+  user: any
+  subscriber: any
+  lesson: Exercise
+  module: SanityDocument
+  section?: SanityDocument
+}
+
+function defineRulesForPurchases(viewerAbilityInput: ViewerAbilityInput) {
+  const {can, rules} = new AbilityBuilder(AppAbility)
+  const {section, module, lesson, subscriber, user} = viewerAbilityInput
+
+  const exercises = section ? section.exercises : module.exercises
+  const isFirstLesson =
+    lesson._type === 'exercise' && lesson._id === exercises?.[0]._id
+
+  const hasVideo = Boolean(lesson.muxPlaybackId)
+
+  console.log({module})
+  if (module.moduleType === 'tutorial') {
+    if (subscriber || (!section && hasVideo)) {
+      can('view', 'Content')
+    }
+  }
+
+  if (isFirstLesson && hasVideo) {
+    can('view', 'Content')
+  }
+
+  if (module.moduleType === 'workshop') {
+    if (user && hasVideo) {
+      can('view', 'Content')
+    }
+  }
+
+  return rules
+}
+
+export function getCurrentAbility(
```

Comment:
we have this in the ability package as well, but I wanted to fiddle with it so moved it in here to diff after and merge back